node/declaration.ts

```diff
@@ -30,19 +30,44 @@ export const getDeclarationsFromConfig = (
   // a function configuration object, we replace the path because that object
   // takes precedence.
   for (const declaration of [...tomlDeclarations, ...deployConfig.declarations]) {
-    const config = functionsConfig[declaration.function] ?? {}
+    const config = functionsConfig[declaration.function]
+
+    // If no config is found, add the declaration as is
+    if (!config) {
+      declarations.push(declaration)
+
+      // If we have path specified create a declaration for each path
+    } else if (config.path?.length) {
+      // eslint-disable-next-line max-depth
+      if (!Array.isArray(config.path)) config.path = [config.path]
+
+      config.path.forEach((path) => {
+        declarations.push({ ...declaration, ...config, path })
+      })
+
+      // If a config was specified, add it to the declaration
+    } else {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { path, ...rest } = config
+      declarations.push({ ...declaration, ...rest })
+    }
 
     functionsVisited.add(declaration.function)
-    declarations.push({ ...declaration, ...config })
   }
 
   // Finally, we must create declarations for functions that are not declared
   // in the TOML at all.
   for (const name in functionsConfig) {
-    const { path, ...config } = functionsConfig[name]
+    const { ...config } = functionsConfig[name]
+    let { path } = functionsConfig[name]
 
+    // If we have path specified create a declaration for each path
     if (!functionsVisited.has(name) && path) {
-      declarations.push({ ...config, function: name, path })
+      if (!Array.isArray(path)) path = [path]
```

Comment:
Same suggestion as above:

```suggestion
      const paths = Array.isArray(path) ? path : [path]
```